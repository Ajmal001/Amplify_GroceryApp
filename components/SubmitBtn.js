import React from "react";
import { Text, TouchableOpacity, StyleSheet } from "react-native";
import { blue, grey } from "../utils/helpers";
import PropTypes from "prop-types";

export default function SubmitBtn({
  title,
  onPress,
  disabled = false,
  style = {},
}) {
  return (
    <TouchableOpacity
      style={
        disabled
          ? [styles.btn, style, { backgroundColor: "#c7c5bf" }]
          : [styles.btn, style]
      }
      onPress={onPress}
      disabled={disabled}
    >
      <Text style={styles.btnText}> {title} </Text>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  btn: {
    backgroundColor: blue,
    borderRadius: 10,
    marginTop: 10,
    paddingTop: 15,
    paddingBottom: 15,
    marginLeft: 100,
    marginRight: 100,
  },
  btnText: {
    color: "#fff",
    textAlign: "center",
  },
});

SubmitBtn.propTypes = {
  title: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
  style: PropTypes.object,
  disabled: PropTypes.bool,
};
