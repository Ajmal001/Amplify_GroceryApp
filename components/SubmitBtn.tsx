import React from "react";
import { Text, TouchableOpacity, StyleSheet } from "react-native";
import { mainColor, secondaryColor } from "../utils/helpers";


interface Props {
  title: string;
  onPress: () => void;
  style?: object,
  disabled?: boolean;
}


const SubmitBtn: React.FC<Props> = ({
  title,
  onPress,
  disabled = false,
  style = {},
}) => {

  return (
    <TouchableOpacity
      style={
        disabled
          ? [styles.btn, style, { backgroundColor: "#c7c5bf" }]
          : [styles.btn, style]
      }
      onPress={onPress}
      disabled={disabled}
    >
      <Text style={styles.btnText}> {title} </Text>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  btn: {
    backgroundColor: mainColor,
    borderRadius: 10,
    marginTop: 10,
    paddingTop: 15,
    paddingBottom: 15,
    marginLeft: 100,
    marginRight: 100,
  },
  btnText: {
    color: secondaryColor,
    textAlign: "center",
  },
});

export default SubmitBtn;