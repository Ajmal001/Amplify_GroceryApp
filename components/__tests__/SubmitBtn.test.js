import React from 'react';
import renderer from 'react-test-renderer';
import SubmitBtn from '../SubmitBtn';

test('renders correctly', () => {
  const tree = renderer.create(<SubmitBtn
    title='title'
    onPress={x=>x} />).toJSON();
  expect(tree).toMatchSnapshot();
});