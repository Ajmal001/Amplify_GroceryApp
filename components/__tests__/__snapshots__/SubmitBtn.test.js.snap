// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#04afea",
      "borderRadius": 10,
      "marginLeft": 100,
      "marginRight": 100,
      "marginTop": 10,
      "opacity": 1,
      "paddingBottom": 15,
      "paddingTop": 15,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "textAlign": "center",
      }
    }
  >
     
    title
     
  </Text>
</View>
`;
